### PR TITLE
development on maintaining bio variable mass balance, not the final push

### PR DESCRIPTION
**Pull request by caglartac**
I have 20 years of simulation with this code. Haven't had an uncontrolled mass increase. I also removed threshold limitations on sediments, but had to increase burial rate by 5 times. I think it is ok to test this in TP2/5.

The code is very mess, but works out of the box. If TP2/5 simulations works fine, I can clean the code later.

### CODE REVIEW

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90
Comment (lilleannette) [inline hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90:782]: @caglartac you should add a comment here as to why this has been commented out

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90
Comment (lilleannette) [inline hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90:706]: @caglartac does not this if-stament mean that "vertical_movement() would not be called if "do_check_state" is set to false?

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90
Comment (caglartac) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90:782]: It is commented out because it will be removed.

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90
Comment (caglartac) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90:706]: .false. means it will not correct variables. The only correction done is after water column sources and sink
This is just a check if the model has infinite values

**Review by lilleannette — CHANGES_REQUESTED** [review]
Caglar, I added a couple of comments to the code.  For me to review it, it would be very helpful if you could put comments in the code to make it easier to identify which blocks of code are there for printing/checking values and which do computations/alterations to the models fields. Annette

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90
Comment (lilleannette) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90:706]: ok, I see it is just a online statement, I thought it was a whole block.

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90
Comment (lilleannette) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90:782]: OK, but if the whole thing will be removed, should you not also remove this comment?
! HIGH VALUES ARE SET TO A BIT LOWER VALUES TO KEEP THE MODEL REPEATEDLY
! EXCEEDING THE 99TH PERCENTILE
! THE SAME THING IS ALSO DONE AFTER THE BOTTOM EXCHANGE

Comment (caglartac) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90:782]: ok

The only change that needs to be reviewed is in lines 1045-1065 which ensures mass balance. The rest is just a print and I will remove them.

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90
Comment (lilleannette) [inline hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90:1048]: @caglar, small thing: you could save the space taken up by "mass_before_check" state by computing the sum in this step - then you can remove the depth dimension.

File: hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90
Comment (lilleannette) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90:782]: @caglartac these lines look good to me, but I will not approve the merging to develop until after you have removed all the print-statements.

Comment (caglartac) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90:782]: Ok I will remove them

Comment (caglartac) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90:782]: @lilleannette I was more thinking that we would use this branch to test the model before pushing to the develop. That is why the prints are there, just in case I need them again if we encounter something with TP2/5

Comment (lilleannette) [reply hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90:782]: @caglartac great, yes, we leave it like this for now, then I may also test it in TP2.

**Review by lilleannette — APPROVED** [review]

**Pull request merged**
